### PR TITLE
ReportIssue: Replace overloads that use properties

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Follow up of #9298 effort to add the last missing overload of `ReportIssue`

This PR replaces all usages that use the properties

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/Extensions/SonarAnalysisContextExtensions.cs [hunk 1]
```diff
@@ -46,7 +46,7 @@ public static class SonarAnalysisContextExtensions
     public static void RegisterCodeBlockStartAction(this SonarAnalysisContext context, Action<SonarCodeBlockStartAnalysisContext<SyntaxKind>> action) =>
         context.RegisterCodeBlockStartAction(CSharpGeneratedCodeRecognizer.Instance, action);
 
-    [Obsolete("Use overload without Diagnostic.Create - the same as in the context class.")]
+    [Obsolete("Use another overload of ReportIssue, without calling Diagnostic.Create")]
     public static void ReportIssue(this SonarCompilationReportingContext context, Diagnostic diagnostic) =>
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Extensions/SonarAnalysisContextExtensions.cs:50]: Can the obsolete methods be removed now?

File: analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantDeclaration.cs [hunk 2]
```diff
@@ -430,9 +424,7 @@ private static void ReportIssueOnRedundantObjectCreation(SonarSyntaxNodeReportin
             var location = node is ObjectCreationExpressionSyntax objectCreation
                 ? objectCreation.CreateLocation(objectCreation.Type)
                 : node.GetLocation();
-            context.ReportIssue(Diagnostic.Create(Rule, location,
-                ImmutableDictionary<string, string>.Empty.Add(DiagnosticTypeKey, redundancyType.ToString()),
-                message));
+            context.ReportIssue(Rule, location, ImmutableDictionary<string, string>.Empty.Add(DiagnosticTypeKey, redundancyType.ToString()), message);
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantDeclaration.cs:427]: The best I could come up with is to make this a bit easier is:
```cs
// Another overload
public void ReportIssue(DiagnosticDescriptor rule, Location location, KeyValuePair<string, string> property, params string[] messageArgs) => ..

// Named argument is needed to disambiguate
context.ReportIssue(Rule, location, property: new(DiagnosticTypeKey, redundancyType.ToString()), message);
```

or using `Dictionary<string, string> properties` and dictionary initializers. Both options are not very pleasent.

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM

File: analyzers/src/SonarAnalyzer.CSharp/Extensions/SonarAnalysisContextExtensions.cs (hunk 1 above)
Comment (martin-strecker-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/Extensions/SonarAnalysisContextExtensions.cs:50]: Discussed on Slack.

Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/Extensions/SonarAnalysisContextExtensions.cs:50]: Yes for the extensions - done in #9356 
No for the base classes, there are too many usages).

File: analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantDeclaration.cs (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantDeclaration.cs:427]: Let's wait for C# 13: #9354

**Pull request merged**